src/components/DataDimension/DataTypesSelector.js

```diff
@@ -1,32 +1,31 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import Select from '@material-ui/core/Select'
-import MenuItem from '@material-ui/core/MenuItem'
-import InputLabel from '@material-ui/core/InputLabel'
 import i18n from '@dhis2/d2-i18n'
+import { SingleSelectField, SingleSelectOption } from '@dhis2/ui-core'
 
 import { dataTypes } from '../../modules/dataTypes'
-import { styles } from './styles/DataTypesSelector.style'
+import styles from './styles/DataTypesSelector.style'
 
 export const DataTypes = ({ currentDataType, onDataTypeChange }) => (
-    <div style={styles.container}>
-        <InputLabel style={styles.titleText}>{i18n.t('Data Type')}</InputLabel>
-        <Select
-            value={currentDataType}
-            onChange={event => onDataTypeChange(event.target.value)}
-            disableUnderline
-            SelectDisplayProps={{ style: styles.dropDown }}
+    <div className="container">
+        <SingleSelectField
+            label={i18n.t('Data Type')}
+            selected={{
```

Comment:
Below a different approach is used, which is maybe better.
If any of `id` or `name` is not defined, the whole `selected` prop won't work because it won't match any option in the select.

Reply:
As this will change for ui-core v5 we'll wait with this until we refactor for that breaking change instead.